plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/XenServerStorageProcessor.java

```diff
@@ -96,6 +96,14 @@ public XenServerStorageProcessor(final CitrixResourceBase resource) {
         hypervisorResource = resource;
     }
 
+    protected String getSRNameLabel(final PrimaryDataStoreTO primaryStore) {
+        if (Storage.StoragePoolType.PreSetup.equals(primaryStore.getPoolType()) &&
+                !primaryStore.getPath().contains(primaryStore.getUuid())) {
+            return  primaryStore.getPath().replace("/", "");
```

Comment:
as above. Make sure it is only the last one being replaced.

Reply:
also why is this method double implemented? here and in CitrixHelper? <quote style='ominous'>there can be only one</quote>

Reply:
`StorageFilerTO`, `PrimaryDataStoreTO` and `StoragePool` different classes

Reply:
Yes, @shwstppr but different classes implementing the exact same doesn't like right to me. It seems to me
1. the method should be a utility (easy dirty quick solution) or
2. the storage path should have it's own class that is referred from all three users of the path.